indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java

```diff
@@ -151,7 +155,9 @@ public IndexTask(
       @JsonProperty("resource") TaskResource taskResource,
       @JsonProperty("spec") IndexIngestionSpec ingestionSchema,
       @JacksonInject ObjectMapper jsonMapper,
-      @JsonProperty("context") Map<String, Object> context
+      @JsonProperty("context") Map<String, Object> context,
+      @JacksonInject IndexMerger indexMerger,
+      @JacksonInject IndexIO indexIO
```

Comment:
Injecting IndexIO and IndexMerger might be a problem for external libraries relying on those classes purely for serialization and de-serialization, such as tranquility. I wonder if it would be possible to separate the parts necessary for serde from the parts needed to run the task.